### ISSUE TITLE
elasticsearch exporter

### ISSUE THREAD

**Issue by Vishnuvardhan-9** [open-telemetry/opentelemetry-collector-contrib#11037]
Hi Team, 

I am trying to configure otelcol-contrib with elasticsearchexporter and seeing below certificate error though tls insecure set to true.  I tested with logstash/filebeat and i don't see any issue with them connecting and writing to elasticsearch.  I have attached elasticsearch exporter/logstash configs inline. Let me know if you need any more details. 

Version : github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/elasticsearchexporter@v0.51.

Error in logs:

Jun 14 16:50:18 otelcol-contrib[30109]: 2022-06-14T16:50:18.632Z        error        elasticsearchexporter@v0.51.0/exporter.go:287        Bulk indexer error: flush: x509: cannot validate certificate for <IP> because it doesn't contain any IP SANs        {"kind": "exporter", "name": "elasticsearch"}

otelcol-contrib config:

exporters:
  elasticsearch:
    endpoints: [https://<hostname>:9200]
    timeout: 2m
    index: syslog-index
    pipeline: mypipeline
    user: XXXXXXX
    password: XXXXXXX
    tls:
      insecure: true
    discover:
      on_start: true
    flush:
      bytes: 10485760
    retry:
      max_requests: 5

Logstash config:
        opensearch {
                hosts => [
                                                        "https://<hostname>:9200"                                    ]
                ssl => true
                ssl_certificate_verification => false
                user => "XXXXXX"
                password => "XXXXXXX"
                index => "syslog-%{+YYYY.MM.dd}"
        }

**Comment by codeboten** [comment]
Pinging codeowners @urso @faec @blakerouse could you have a look and assign a priority label for this issue. Thanks!

**Comment by JaredTan95** [comment]
I think the logs are already very clear about the problem:

```bash
Bulk indexer error: flush: x509: cannot validate certificate for because it doesn't contain any IP SANs
```

**Comment by JaredTan95** [comment]
Maybe you can try more `tls` config as https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticsearchexporter#tls-settings mentioned.

**Comment by Vishnuvardhan-9** [comment]
Thanks, new tls config insecure_skip_verify helped to get rid of the certificate error. Now I am seeing a new error related to opensearch compatibility. Little background on what we are trying. We are trying to configure elasticsearch Exporter to work with Opensearch endpoint. We have enabled a special variable in opensearch to avoid compatibility issues with ingest tools and seems it is not working. 

We followed the instruction given in below opensearch url and add the below variable in config to avoid compatibility with ingest tools. 
compatibility.override_main_response_version: true

Url:
https://opensearch.org/docs/latest/clients/agents-and-ingestion-tools/index/

Error seeing in logs:
otelcol-contrib[23109]: 2022-06-17T16:01:13.804Z        error        elasticsearchexporter@v0.51.0/exporter.go:287        Bulk indexer error: flush: the client noticed that the server is not Elasticsearch and we do not support this unknown product        {"kind": "exporter", "name": "elasticsearch"}
